### ISSUE TITLE
Redis context deadline exceeded

### ISSUE THREAD

**Issue by MAHDTech** [oauth2-proxy/oauth2-proxy#2331]
## Expected Behavior

I'm using the KubeApps helm chart which bundles `oauth2-proxy` v7.5.1

I have an existing Redis v7.2.1 deployment and I'm attempting to connect `oauth2-proxy` to it.

## Current Behavior

The following error is shown and the Pod goes into `CrashLoopBackOff`

```bash
kubeapps/kubeapps-768c75d857-nvp72[auth-proxy]: [2023/11/22 00:29:17] [main.go:54] invalid configuration:
kubeapps/kubeapps-768c75d857-nvp72[auth-proxy]:   unable to set a redis initialization key: context deadline exceeded
kubeapps/kubeapps-768c75d857-nvp72[auth-proxy]:   unable to delete the redis initialization key: context deadline exceeded
```

## Steps to Reproduce (for bugs)

I have performed the following steps.

1. Deployed `redis` helm chart in a standalone config with no authentication for testing
1. Deployed the `kubeapps` helm chart which bundles `oauth2-proxy` v7.5.1
1. Passed these redis related flags to `oauth2-proxy` via the helm chart

```yaml
- --session-store-type=redis
- --redis-connection-url=redis://kubeapps-redis-authproxy-master.kubeapps.svc.cluster.local:6379
- --redis-insecure-skip-tls-verify=true
- --redis-use-cluster=false
- --redis-connection-idle-timeout=15
```

## Context

I'm in an environment with a large number of groups, where the Cookie size is > 4kb. This is the reason I've started going down the Redis path.

**Questions:** 

- I only have a small number of groups who need access to use the application, is there a way I can filter these groups and reduce the Cookie size to allieviate the need for redis at all?
- Is the version of Redis I am using (v7.2.1) compatible with oauth2-proxy (v7.5.1)
- Have I missed any required flags to get this to work?

## Your Environment

- Version used:  Tested both v7.4.0 and v7.5.1 with the same error.
- KubeApps Helm Chart which deploys oauth2-proxy version 7.5.1
- The OIDC IdP that is used is VMware Workspace ONE Access

I can successfully attach a debug container to the Pod where `oauth2-proxy` is running and connect to redis successfully.

```bash
kubectl debug kubeapps-5dc4895b7d-ttnlk -it --namespace kubeapps --image=redis -- /bin/sh

# redis-cli -h kubeapps-redis-authproxy-master.kubeapps.svc -p 6379 PING
PONG

# redis-cli -h kubeapps-redis-authproxy-master.kubeapps.svc -p 6379 INFO SERVER
# Server
redis_version:7.2.1
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:ed7eac84dd41e25a
redis_mode:standalone
os:Linux 4.19.283-3.ph3 x86_64
arch_bits:64
monotonic_clock:POSIX clock_gettime
multiplexing_api:epoll
atomicvar_api:c11-builtin
gcc_version:10.2.1
process_id:1
process_supervised:no
run_id:c3409af248271a3e906cba787249dddd539d8bbd
tcp_port:6379
server_time_usec:1700612623625771
uptime_in_seconds:980
uptime_in_days:0
hz:10
configured_hz:10
lru_clock:6113807
executable:/redis-server
config_file:
io_threads_active:0
listener0:name=tcp,bind=*,bind=-::*,port=6379
```

**Comment by MAHDTech** [comment]
Although the error wasn't particulary helpful, I ended up getting there.

The issue was this

- I was using the HTTP scheme `redis://`
- I had the `redis-insecure-skip-tls-verify` flag specified

When I removed the following flag, no further redis errros appeared in the logs.

```bash
- --redis-insecure-skip-tls-verify=true
```

I had incorrectly assumed as I was testing without creating any certs, I needed to ignore them, but TLS is not enabled with the `redis://` scheme, would instead be `rediss://`

Cheers.
